### PR TITLE
eig extension

### PR DESCRIPTION
**Pull request by zhumakhan**
#6180

### CODE REVIEW

**Comment by zhumakhan** [comment]
Test is not added due to non-uniqueness of eigenvectors: https://pytorch.org/docs/stable/generated/torch.linalg.eig.html: "The returned eigenvectors are normalized to have norm 1. Even then, the eigenvectors of a matrix are not unique, nor are they continuous with respect to A. Due to this lack of uniqueness, different hardware and software may compute different eigenvectors."

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi. Thanks for the PR. We will still require a test. Yes the values may not be unique but many other attributes of the output sequence of arrays like dtype etc. to be checked. To ensure values are not tested for their correctness, the argument `test_values` can be set to false in `test_function`. 
🙂
Let me know if the above doesn't make sense or if you have any further queries. Thank you

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi. Apologies for the delay. This PR was opened and closed several times and I got a bit mixed up. Your PR seems good but there are several conflicts to resolve. Could you resolve these kindly and ask for a review 🙂. Feel free to ask any questions. Thank @zhumakhan

**Comment by zhumakhan** [comment]
@Ishticode conflicts are resolved, thanks a lot for you comments

File: ivy/functional/backends/tensorflow/experimental/linear_algebra.py [hunk 1]
```diff
@@ -47,3 +47,18 @@ def kron(
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Union[tf.Tensor, tf.Variable]:
     return tf.experimental.numpy.kron(a, b)
+
+
+"""
+Natively supports float32, float64, complex64, complex128 data types.
```
Comment (Ishticode) [inline ivy/functional/backends/tensorflow/experimental/linear_algebra.py:53]: we don't really require these comments

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 2]
```diff
@@ -217,3 +217,49 @@ def test_kron(
         a=x[0],
         b=x[1],
     )
+
+
+"""eig. Values are not tested since eigenvectors are not unique
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:222]: These comments are not required

File: ivy/container/experimental/linear_algebra.py [hunk 3]
```diff
@@ -170,3 +170,62 @@ def kron(
...
+    def static_eig(
+        x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        /,
+        *,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
```
Comment (Ishticode) [inline ivy/container/experimental/linear_algebra.py:184]: Could we add a docstring for this too... with some examples : )

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @zhumakhan 
Thanks for the update. Could you take care of the following comments.
